### PR TITLE
[full-ci][tests-only] add test to download preview of an image and text file with preview param set to zero

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds tests to check for preview of an image and text file by setting preview query param to zero.

## Related Issue

- Related to: https://github.com/owncloud/ocis/issues/9949

## How Has This Been Tested?

- locally
- CI
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 1]
```diff
@@ -315,3 +315,31 @@ Feature: previews of files downloaded through the webdav API
       | old              |
       | new              |
       | spaces           |
+
+
+  Scenario Outline: download previews of an image with preview set to zero
+    Given using <dav-path-version> DAV path
+    And user "Alice" has uploaded file "filesForUpload/testavatar.jpg" to "/testavatar.jpg"
+    When user "Alice" downloads the preview of "/testavatar.jpg" with width "32" and height "32" with preview "0" using the WebDAV API
```
Comment (amrita-shrestha) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:323]: maybe we can use table to pass preview queries

Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:323]: this should be the only time we gonna use this step
```suggestion
    When user "Alice" tries to download the preview of "/testavatar.jpg" with width "32" and height "32" and preview set to 0 using the WebDAV API
```

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 2]
```diff
@@ -315,3 +315,31 @@ Feature: previews of files downloaded through the webdav API
       | old              |
       | new              |
       | spaces           |
+
+
+  Scenario Outline: download previews of an image with preview set to zero
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:320]: we set `0`
```suggestion
  Scenario Outline: try to download preview of an image with preview set to 0
```

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 3]
```diff
@@ -315,3 +315,31 @@ Feature: previews of files downloaded through the webdav API
...
+    Then the HTTP status code should be "200"
+    And the downloaded image should be "1240" pixels wide and "648" pixels high
+    Examples:
+      | dav-path-version |
+      | old              |
+      | new              |
+      | spaces           |
+
+
+  Scenario Outline: download previews of a text file with preview set to zero
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:333]:
```suggestion
  Scenario Outline: try to download preview of a text file with preview set to 0
```

File: tests/acceptance/bootstrap/WebDav.php [hunk 4]
```diff
@@ -3691,6 +3691,40 @@ public function downloadPreviewOfFiles(string $user, string $path, string $width
...
+	 *
+	 * @return void
+	 */
+	public function userDownloadsThePreviewOfWithPreviewZero(string $user, string $path, string $width, string $height, int $preview) {
+		$user = $this->getActualUsername($user);
+		$urlParameter = [
+			'x' => $width,
+			'y' => $height,
+			'forceIcon' => '0',
+			'preview' => $preview
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/WebDav.php:3711]:
```suggestion
			'preview' => 0
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/bootstrap/WebDav.php [hunk 5]
```diff
@@ -3691,6 +3691,39 @@ public function downloadPreviewOfFiles(string $user, string $path, string $width
...
+	 *
+	 * @return void
+	 */
+	public function userDownloadsThePreviewOfWithPreviewZero(string $user, string $path, string $width, string $height) {
+		$user = $this->getActualUsername($user);
+		$urlParameter = [
+			'x' => $width,
+			'y' => $height,
+			'forceIcon' => '0',
+			'preview' => '0'
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/WebDav.php:3710]:
```suggestion
			'y' => $height,
			'preview' => '0'
```

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 6]
```diff
@@ -315,3 +315,31 @@ Feature: previews of files downloaded through the webdav API
       | old              |
       | new              |
       | spaces           |
+
+
+  Scenario Outline: try to download preview of an image with preview set to 0
+    Given using <dav-path-version> DAV path
+    And user "Alice" has uploaded file "filesForUpload/testavatar.jpg" to "/testavatar.jpg"
+    When user "Alice" tries to download the preview of "/testavatar.jpg" with width "32" and height "32" with preview set to 0 using the WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:323]:
```suggestion
    When user "Alice" tries to download the preview of "/testavatar.jpg" with width "32" and height "32" and preview set to 0 using the WebDAV API
```

File: tests/acceptance/features/coreApiWebdavPreviews/previews.feature [hunk 7]
```diff
@@ -315,3 +315,29 @@ Feature: previews of files downloaded through the webdav API
...
+      | new              |
+      | spaces           |
+
+
+  Scenario Outline: try to download preview of a text file with preview set to 0
+    Given using <dav-path-version> DAV path
+    And user "Alice" has uploaded file with content "to preview" to "textfile.txt"
+    When user "Alice" tries to download the preview of "textfile.txt" with width "32" and height "32" and preview set to 0 using the WebDAV API
+    Then the HTTP status code should be "200"
+    And for user "Alice" the content of the file "/textfile.txt" of the space "Personal" should be "to preview"
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiWebdavPreviews/previews.feature:338]: This step downloads and checks the content, the behavior is not suitable here.
We need to check the content from the saved response
```suggestion
    check content from the response
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

**Review by saw-jan — APPROVED** [review]
LGTM :+1:
